converter/shieldhit/parser.py

```diff
@@ -207,6 +226,37 @@ def _get_detector_by_uuid(self, detect_uuid: str) -> Optional[str]:
 
         raise ValueError(f"No detector with uuid {detect_uuid}")
 
+    def _parse_quantity_settings(self, quantity_dict: dict) -> dict or None:
+        """Parses settings from the input json into the quantity settings property"""
+
+        def create_name_from_settings() -> str:
+            """Create a name for the quantity from its settings."""
```

Comment:
Add here some good comment what this method is doing. its hard to read its logic.